pkg/errors/errors.go

```diff
@@ -16,10 +16,16 @@ type FlyteAdminError interface {
 	Error() string
 	Code() codes.Code
 	GRPCStatus() *status.Status
-	WithDetails(details *admin.EventFailureReason) (FlyteAdminError, error)
+	WithDetails(details ProtoMessager) (FlyteAdminError, error)
```

Comment:
I believe there is a `proto.Message` interface that looks awfully similar. maybe take a look at that?

Reply:
@EngHabu you're right! Found it and plugged it int. Thanks